_docs/javascript-api.md

```diff
@@ -1517,6 +1517,27 @@ Java.perform(function () {
 });
 {% endhighlight %}
 
+**Note:** In case you are using on an older version of Frida (for whatsoever reasons), early instrumentation may still be achieved using `Java.performNow()` instead of `Java.perform()`
```

Comment:
I guess I got it all wrong. Would you mind explaining what's really going on in the below example to help clarify my concepts around Java.perform() and Java.performNow()

Example:
I discovered Java.performNow() while doing an internal pen test. It was an android app that was doing some in app crypto stuff. Consequently, we would see only cipher text in BURP logs and as the first step we were supposed to break the crypto itself to be able to get the plain text requests from the app in the first place. 

In the first version of the app (the first round of our test) we found a class in the decompiled apk that seemed to be doing all the crypto. We hooked each of the methods in that class and were able to decrypt the traffic. 

However, in the second version (round 2 of our test) the app was heavily obfuscated with Dexguard. Hence, unlike version 1, we could not find any direct reference to a class that was doing the crypto. It was then that we tried hooking the basic Java crypto libraries and methods itself to see if that would help. That's when we tried hooking Java's KeyPair.getPrivate()

Now here's what I do not understand. KeyPair.getPrivate() would definitely be called from within a class in the app. Which means that only once this class (in the app) is loaded, will the    KeyPair.getPrivate() be loaded. So essentially, it is not a 'system APIs called before the app's classes are loaded'. Then how does Java.performNow() help here the way you explained above ?
Please feel free to correct me or give pointers/references in case my basic understanding itself is incorrect.